### PR TITLE
Move BOM info to right after Quick Start

### LINKED ISSUE #3537: Move BOM usage up to the front
Currently BOM info is towards the bottom of the doc and easily missed.

It should be on the top, either in the quickstart, or immediately after that.

### PR DESCRIPTION
**Pull request by yihanzhen**
Fixes #3537

### CODE REVIEW

**Comment by hzyi-google** [comment]
cc/ @saturnism

**Comment by hzyi-google** [comment]
PTAL @garrettjonesgoogle

File: README.md [hunk 1]
```diff
@@ -337,6 +298,63 @@ a higher priority.
...
+        <groupId>com.google.cloud</groupId>
+        <artifactId>google-cloud-bom</artifactId>
+        <version>0.57.0-alpha</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+```
+[//]: # ({x-version-update-end})
```
Comment (garrettjonesgoogle) [inline README.md]: The stuff above just repeats what is in the quick start section, and the stuff below is for version combinations before the BOM. I think I'd prefer to remove the repeated stuff above and rename this section to something like `Old version combinations`. (Also change the text at the end of the quick start that refers to this section.)

**Comment by hzyi-google** [comment]
PTAL

**Pull request merged**